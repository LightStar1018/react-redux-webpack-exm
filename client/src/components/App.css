.App {
    text-align: center;
    display: flex;
    flex-flow: column nowrap;
    justify-content: center;
    align-items: center;
    background-color: rgb(190, 179, 190);
    height: 100vh;
  }
  h1 {
    margin-bottom: 30px;
  }
  .form-data {
    display: flex;
    flex-flow: column;
    width: fit-content;
    align-items: center;
    justify-content: center;
  }
  input[type="text"] {
    font-size: 1.2rem;
    margin: 10px;
    text-align: center;
  }
  input[type="text"]:disabled {
    background-color: #fff;
  }
  label {
    font-size: 1.3rem;
  }
  pre {
    white-space: pre-wrap;
    width: 50%;
  }
  .input-container {
    position: relative;
  }
  .icon-container {
    position: absolute;
    right: 10px;
    top: calc(50% - 10px);
  }
  .loader {
    position: relative;
    height: 20px;
    width: 20px;
    display: inline-block;
    animation: around 5.4s infinite;
  }
  @keyframes around {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
  
  .loader::after,
  .loader::before {
    content: "";
    background: white;
    position: absolute;
    display: inline-block;
    width: 100%;
    height: 100%;
    border-width: 2px;
    border-color: #333 #333 transparent transparent;
    border-style: solid;
    border-radius: 20px;
    box-sizing: border-box;
    top: 0;
    left: 0;
    animation: around 0.7s ease-in-out infinite;
  }
  
  .loader::after {
    animation: around 0.7s ease-in-out 0.1s infinite;
    background: transparent;
  }