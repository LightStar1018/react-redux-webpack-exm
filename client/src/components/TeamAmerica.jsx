import React, { Component,useState } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import '../components/App.css';

// Actions being imported
import { getTeam } from '../actions/planTeamActions';

// This is my React class that has the basic code to get my example to render
// I had to export the class like this so Jest could read it
export class TeamAmerica extends Component {
  constructor(props) {
    super(props)
    // this.zipcode={zipcode:""}
  }
  componentDidMount() {
    this.props.getTeam();
  }

  render() {
    // const initialCityState = { city: "", state: "" };
    // const [cityState, setCityState] = useState(initialCityState);
    // const [zipcode, setZipcode] = useState("");
    return (
      <div>
        <h1>City/State Lookup Tool</h1>
      <form action="" className="form-data">
        <label htmlFor="zip">Type Zip Code Here</label>
        <input
          className="zip"
       
          placeholder="XXXXX"
          type="text"
          name="zip"
          id="zip"
        />
        <label htmlFor="city">City</label>
        <input
          className={`city`}
          
          type="text"
          name="city"
          disabled
          id="city"
        />
        <label htmlFor="state">State</label>
        <input
          className={`state`}
         
          type="text"
          name="state"
          disabled
          id="state"
        />
      </form>
      <pre>
        <code>
          {/* {JSON.stringify({
            zipcode: zipcode,
            city: cityState.city,
            state: cityState.state,
          })} */}
        </code>
      </pre>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    team: state.planTeam.team,
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ getTeam }, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(TeamAmerica);
