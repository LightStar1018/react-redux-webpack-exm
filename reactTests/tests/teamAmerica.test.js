import React from 'react';
import renderer from 'react-test-renderer';
import { TeamAmerica } from './../../client/src/components/TeamAmerica';

// I had to copy paste the API json into this file to fix a bug
const team = {
  presidents: [
    {
      team: 'Presidents',
      team_id: '1',
      first_name: 'George',
      full_name: 'George Washington',
      last_name: 'Washington',
      role: 'First President of the United State of America.',
    },
    {
      team: 'Presidents',
      team_id: '2',
      first_name: 'Abraham',
      full_name: 'Abraham Lincoln',
      last_name: 'Lincoln',
      role: 'President during the Civil War.',
    },
    {
      team: 'Presidents',
      team_id: '3',
      first_name: 'Franklin D.',
      full_name: 'Franklin D. Roosevelt',
      last_name: 'Roosevelt',
      role: 'President during The Great Depression and WW2. Only president to be elected 4 times.',
    },
  ],
  client: [
    {
      team: 'US People',
      team_id: '2',
      first_name: 'Joe',
      full_name: 'Joe Sixpack',
      last_name: 'Sixpack',
      role: 'Average American',
    },
  ],
};

// I was getting an error that getTeam was not a function, so I had to turn it into a function in the test by doing this fat arrow
const getTeam = () => {};

// This is the actual test written for Jest
it('test to see if the team renders correctly', () => {
  const tree = renderer
  .create(<TeamAmerica getTeam={getTeam} team={[team.presidents]} />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});

