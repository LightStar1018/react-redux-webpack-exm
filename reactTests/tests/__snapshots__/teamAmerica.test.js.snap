// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to see if the team renders correctly 1`] = `
<div>
  <h1>
    Team America
  </h1>
  <div>
    <h2>
      <div>
        Presidents
      </div>
      <div />
      <div />
    </h2>
    <ul>
      <li>
        George Washington
         - 
        First President of the United State of America.
         
      </li>
    </ul>
    <ul>
      <li>
        Abraham Lincoln
         - 
        President during the Civil War.
         
      </li>
    </ul>
    <ul>
      <li>
        Franklin D. Roosevelt
         - 
        President during The Great Depression and WW2. Only president to be elected 4 times.
         
      </li>
    </ul>
  </div>
</div>
`;
